tests/Operation/ExplainFunctionalTest.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+namespace MongoDB\Tests\Operation;
+
+use MongoDB\Operation\Count;
+use MongoDB\Operation\Distinct;
+use MongoDB\Operation\Explain;
+use MongoDB\Operation\FindAndModify;
+use MongoDB\Operation\InsertMany;
+
+class ExplainFunctionalTest extends FunctionalTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+        if (version_compare($this->getServerVersion(), '3.0.0', '<')) {
+            $this->markTestSkipped('Explain is not supported');
+        }
+    }
+
+    public function testCountAllPlansExecution()
+    {
+        $insertMany = new InsertMany($this->getDatabaseName(), $this->getCollectionName(), [
+            ['x' => 0],
+            ['x' => 1],
+            ['x' => 2],
+            ['y' => 3]
+        ]);
+        $insertMany->execute($this->getPrimaryServer());
+
+        $operation = new Count($this->getDatabaseName(), $this->getCollectionName(), ['x' => ['$gte' => 1]], []);
+        $explainOperation = new Explain($this->getDatabaseName(), $operation, ['verbosity' => Explain::VERBOSITY_ALL_PLANS, 'typeMap' => ['root' => 'array']]);
+        $result = $explainOperation->execute($this->getPrimaryServer());
+
+        $this->assertTrue(array_key_exists('queryPlanner', $result));
```

Comment:
[`assertArrayHasKey()`](https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.assertArrayHasKey) can be used here. In addition to being more concise, it also also PHPUnit to use a more intelligent error message if the assertion fails (e.g. "array didn't have the key we expected" instead of "expected true and got false").